model/project_parser.go

```diff
@@ -569,18 +594,36 @@ type GetProjectOpts struct {
 	Token      string
 }
 
-func GetProjectFromFile(ctx context.Context, opts GetProjectOpts) (*Project, *ParserProject, error) {
-	configFile, err := thirdparty.GetGithubFile(ctx, opts.Token, opts.Ref.Owner, opts.Ref.Repo, opts.RemotePath, opts.Revision)
+func retrieveFile(ctx context.Context, opts GetProjectOpts) ([]byte, error) {
+	conf, err := evergreen.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get evergreen configuration")
+	}
+	ghToken, err := conf.GetGithubOauthToken()
 	if err != nil {
-		return nil, nil, errors.Wrapf(err, "error fetching project file for '%s' at '%s'", opts.Ref.Id, opts.Revision)
+		return nil, errors.Wrap(err, "can't get Github OAuth token from configuration")
 	}
+	configFile, err := thirdparty.GetGithubFile(ctx, ghToken, opts.Ref.Owner, opts.Ref.Repo, opts.RemotePath, opts.Revision)
```

Comment:
Revision will default to main/master if not specified, which is usually fine. If RemotePath isn't given, we should add handling to get it from opts.Ref